modules/dnn/test/test_onnx_importer.cpp

```diff
@@ -2745,6 +2745,10 @@ TEST_P(Test_ONNX_layers, Expand_shape_model4) {
     testONNXModels("test_expand_shape_model4", pb, 0, 0, false, true, 1);
 }
 
+TEST_P(Test_ONNX_layers, Attention) {
+    testONNXModels("attention");
+}
```

Comment:
This test fails with disabled `AttentionSubGraph`. Is that expected?

Reply:
Benchmark `w/o Attention` is done with disabled `AttentionSubGraph`. It should work. What is the error? Accuracy?

Reply:
Yes, this is about the  accuracy test only:
```
[ RUN      ] Test_ONNX_layers.Attention/1, where GetParam() = OCV/CPU
[ INFO:0@0.062] global onnx_importer.cpp:797 populateNet DNN/ONNX: loading ONNX v9 model produced by ''. Number of nodes = 34, initializers = 0, inputs = 1, outputs = 1
[ INFO:0@0.062] global onnx_importer.cpp:698 parseOperatorSet DNN/ONNX: ONNX opset version = 19
[ INFO:0@0.062] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 1 inputs and 1 outputs: [Transpose]:(onnx_node!n0) from domain='ai.onnx'
[ INFO:0@0.062] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n1) from domain='ai.onnx'
[ INFO:0@0.062] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [MatMul]:(onnx_node!n2) from domain='ai.onnx'
[ INFO:0@0.062] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n3) from domain='ai.onnx'
[ INFO:0@0.062] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [Add]:(onnx_node!n4) from domain='ai.onnx'
[ INFO:0@0.062] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n5) from domain='ai.onnx'
[ INFO:0@0.062] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n6) from domain='ai.onnx'
[ INFO:0@0.062] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n7) from domain='ai.onnx'
[ INFO:0@0.062] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 4 inputs and 1 outputs: [Slice]:(onnx_node!n8) from domain='ai.onnx'
[ INFO:0@0.062] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n9) from domain='ai.onnx'
[ INFO:0@0.062] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [Reshape]:(onnx_node!n10) from domain='ai.onnx'
[ INFO:0@0.062] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 1 inputs and 1 outputs: [Transpose]:(onnx_node!n11) from domain='ai.onnx'
[ INFO:0@0.062] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n12) from domain='ai.onnx'
[ INFO:0@0.062] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [Div]:(onnx_node!n13) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n14) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n15) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n16) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 4 inputs and 1 outputs: [Slice]:(onnx_node!n17) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n18) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [Reshape]:(onnx_node!n19) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 1 inputs and 1 outputs: [Transpose]:(onnx_node!n20) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [MatMul]:(onnx_node!n21) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 1 inputs and 1 outputs: [Softmax]:(onnx_node!n22) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n23) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n24) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n25) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 4 inputs and 1 outputs: [Slice]:(onnx_node!n26) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n27) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [Reshape]:(onnx_node!n28) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 1 inputs and 1 outputs: [Transpose]:(onnx_node!n29) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [MatMul]:(onnx_node!n30) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 1 inputs and 1 outputs: [Transpose]:(onnx_node!n31) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 0 inputs and 1 outputs: [Constant]:(onnx_node!n32) from domain='ai.onnx'
[ INFO:0@0.063] global onnx_importer.cpp:971 handleNode DNN/ONNX: processing node with 2 inputs and 1 outputs: [Reshape]:(onnx_node!n33) from domain='ai.onnx'
/home/dkurtaev/opencv/modules/dnn/test/test_common.impl.hpp:76: Failure
Expected: (normL1) <= (l1), actual: 27.5616 vs 1e-05
attention  |ref| = 18.29339599609375
/home/dkurtaev/opencv/modules/dnn/test/test_common.impl.hpp:79: Failure
Expected: (normInf) <= (lInf), actual: 4557.7 vs 0.0001
attention  |ref| = 18.29339599609375
[  FAILED  ] Test_ONNX_layers.Attention/1, where GetParam() = OCV/CPU (24 ms)
```

Reply:
Ok, interesting, let me investigate it.

Reply:
Problem is confirmed. I did not set `axis` attribute in Softmax operator. ONNX's Softmax have `axis=-1` as default, while dnn's default to 1. I think it is time to make it align with ONNX's.

Check the latest commit in https://github.com/opencv/opencv_extra/pull/1128.

Reply:
It's because onnx operator define `opset<=11` default axis is 1. Some old tests haven't opset value, so I use default 1 if no opset. 

I will use "If no opset, 2d input use 1, nd use -1 as default" to replace this part.

Reply:
Solved in PR: https://github.com/opencv/opencv/pull/24613